tests/src/rules/prefer-default-export.js

```diff
@@ -7,153 +7,329 @@ import { version as tsEslintVersion } from 'typescript-eslint-parser/package.jso
-    }),
-    test({
-      code: `export Memory, { MemoryValue } from './Memory'`,
-      parser: parsers.BABEL_OLD,
-    }),
-
-    // no exports at all
-    test({
-      code: `
-        import * as foo from './foo';`,
-    }),
-
-    test({
-      code: `export type UserId = number;`,
-      parser: parsers.BABEL_OLD,
-    }),
-
-    // issue #653
-    test({
-      code: 'export default from "foo.js"',
-      parser: parsers.BABEL_OLD,
-    }),
-    test({
-      code: 'export { a, b } from "foo.js"',
-      parser: parsers.BABEL_OLD,
-    }),
-    // ...SYNTAX_CASES,
-    test({
-      code: `
-        export const [CounterProvider,, withCounter] = func();;
-      `,
-      parser: parsers.BABEL_OLD,
-    }),
-    // es2022: Arbitrary module namespae identifier names
-    testVersion('>= 8.7', () => ({
-      code: 'let foo; export { foo as "default" };',
-      parserOptions: { ecmaVersion: 2022 },
-    })),
-  ),
-  invalid: [
-    test({
-      code: `
-        export function bar() {};`,
-      errors: [{
-        type: 'ExportNamedDeclaration',
-        message: 'Prefer default export.',
-      }],
-    }),
-    test({
-      code: `
-        export const foo = 'foo';`,
-      errors: [{
-        type: 'ExportNamedDeclaration',
-        message: 'Prefer default export.',
-      }],
-    }),
-    test({
-      code: `
+    options: [{
+      target: 'any',
+    }],
+    errors: [{
+      message: ANY_EXPORT_ERROR_MESSAGE,
+    }],
+  }),
+  test({
+    code: 'export { a, b } from "foo.js"',
+    parser: parsers.BABEL_OLD,
+    options: [{
+      target: 'any',
+    }],
+    errors: [{
+      message: ANY_EXPORT_ERROR_MESSAGE,
+    }],
+  }),
+  test({
+    code: `
         const foo = 'foo';
         export { foo };`,
-      errors: [{
-        type: 'ExportSpecifier',
-        message: 'Prefer default export.',
-      }],
-    }),
-    test({
-      code: `
+    options: [{
+      target: 'any',
+    }],
+    errors: [{
+      message: ANY_EXPORT_ERROR_MESSAGE,
+    }],
+  }),
+  test({
+    code: `
         export const { foo } = { foo: "bar" };`,
-      errors: [{
-        type: 'ExportNamedDeclaration',
-        message: 'Prefer default export.',
-      }],
-    }),
-    test({
-      code: `
+    options: [{
+      target: 'any',
+    }],
+    errors: [{
+      message: ANY_EXPORT_ERROR_MESSAGE,
+    }],
+  }),
+  test({
+    code: `
         export const { foo: { bar } } = { foo: { bar: "baz" } };`,
-      errors: [{
-        type: 'ExportNamedDeclaration',
-        message: 'Prefer default export.',
-      }],
-    }),
-    test({
-      code: `
-        export const [a] = ["foo"]`,
-      errors: [{
-        type: 'ExportNamedDeclaration',
-        message: 'Prefer default export.',
-      }],
-    }),
-  ],
-});
+    options: [{
+      target: 'any',
+    }],
+    errors: [{
+      message: ANY_EXPORT_ERROR_MESSAGE,
+    }],
+  }),
+);
+
+ruleTester.run('prefer-default-export', rule, { valid, invalid });
```

Comment:
can we flip this back to the inline valid/invalids, so the diff isn't so noisy?

Reply:
will do